src/Compilers/CSharp/Portable/Symbols/Metadata/PE/TupleTypeDecoder.cs

```diff
@@ -200,6 +204,52 @@ private TypeSymbol DecodeType(TypeSymbol type)
             }
         }
 
+        private PointerTypeSymbol DecodePointerType(PointerTypeSymbol type)
+        {
+            return type.WithPointedAtType(DecodeTypeInternal(type.PointedAtTypeWithAnnotations));
+        }
+
+        private FunctionPointerTypeSymbol DecodeFunctionPointerType(FunctionPointerTypeSymbol type)
+        {
+            var parameterTypes = ImmutableArray<TypeWithAnnotations>.Empty;
+            var paramsModified = false;
+
+            if (type.Signature.ParameterCount > 0)
+            {
+                var paramsBuilder = ArrayBuilder<TypeWithAnnotations>.GetInstance(type.Signature.ParameterCount);
+
+                for (int i = type.Signature.ParameterCount - 1; i >= 0; i--)
```

Comment:
>for (int i = type.Signature.ParameterCount - 1; i >= 0; i--) [](start = 16, length = 60)

This looks suspicious. Why are we going through parameters in reversed order only to ReverseContents later?